compiler/codegen/codegen.cpp

```diff
@@ -161,6 +161,14 @@ static void legalizeName(Symbol* sym) {
   {
     sym->cname = astr("chpl__", sym->cname);
   }
+
+  // Append number of array dimensions to polly_array_index
+  // It helps Polly Optimizer to select the correct function
+  if (strcmp("polly_array_index",sym->cname) == 0){
```

Comment:
I think this should use sym->name (vs sym->cname) in the strcmp.

Reply:
Done